### PR TITLE
fix(modal): Show/Hide when once prevented

### PR DESCRIPTION
**Pull request by jackmu95**
## Description of Pull Request:
#2270 broke showing/hiding of once prevented modals.
This PR fixes this.

## PR checklist:

**What kind of change does this PR introduce?** (check at least one)
- [x] Bugfix
- [ ] Feature
- [ ] Enhancement to an existing feature
- [ ] ARIA accessibility
- [ ] Documentation update
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)
- [ ] Yes
- [x] No

If yes, please describe the impact:

**The PR fulfills these requirements:**
- [x] It's submitted to the `dev` branch, **not** the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (i.e. `fixes #xxxx[,#xxxx]`, where "xxxx" is the issue number)
- [x] The PR should address only one issue or feature. If adding multiple features or fixing a bug and adding a new feature, break them into separate PRs if at all possible.
- [x] PR titles should following the [**Conventional Commits**](https://www.conventionalcommits.org/) naming convention (i.e. "fix(alert): not alerting during SSR render", "docs(badge): Updated pill examples, fix typos", "chore: fix typo in docs", etc). **This is very important, as the `CHANGELOG` is generated from these messages.**

**If new features/enhancement/fixes are added or changed:**
- [ ] Includes documentation updates (including updating the component's `package.json` for slot and event changes)
- [ ] New/updated tests are included and passing (if required)
- [ ] Existing test suites are passing
- [ ] The changes have not impacted the functionality of other components or directives 
- [ ] ARIA Accessibility has been taken into consideration (does it affect screen reader users or keyboard only users? clickable items should be in the tab index, etc)

**If adding a new feature, or changing the functionality of an existing feature, the PR's description above includes:**
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

### CODE REVIEW

**Comment by jackmu95** [comment]
@tmorehouse Would be nice if we could solve this quickly :)

**Comment by tmorehouse** [comment]
Will just need to make sure `this.is_opening`/`this.is_closing` are falsified out on cancel of show/hide

File: src/components/modal/modal.js [hunk 1]
```diff
@@ -565,8 +565,9 @@ export default {
         relatedTarget: null
       })
       this.emitEvent(showEvt)
+      // Don't show if canceled
       if (showEvt.defaultPrevented || this.is_visible) {
-        // Don't show if canceled
+        this.is_opening = false
```
Comment (tmorehouse) [inline src/components/modal/modal.js]: Good catch.

**Comment by tmorehouse** [comment]
Just thinking about the root listeners that may be added, and as a just-in-case, remove any (with `this.$root.$off(..., this.doShow)`) in `beforeDestroy()` hook

**Comment by jackmu95** [comment]
@tmorehouse Feel free to apply any changes 👍

**Comment by tmorehouse** [comment]
Ok, will add the $off events in before destroy, just in case someone takes the component out of the page (I.e. route change, etc)

**Comment by tmorehouse** [comment]
Added in.. any thing else before merging?

**Comment by jackmu95** [comment]
No, that should be it.

**Comment by tmorehouse** [comment]
thanks for catching this!

**Pull request merged**